src/Raven.Server/ServerWide/Maintenance/ClusterObserver.cs

```diff
@@ -278,6 +280,12 @@ private void LogMessage(string message, Exception e = null, string database = nu
 
                     if (cleanupTombstones && _hasMoreTombstones == false)
                         _lastTombstonesCleanupTimeInTicks = now.Ticks;
+
+                    if (cleanupExpiredCompareExchange)
+                    {
+                        if (await RemoveExpiredCompareExchange(context, now.Ticks) == false)
```

Comment:
No related to this issue and I know we do it all over the place here, but I wonder if the observer need to `await` for anything.. 
cc @ayende

Reply:
We don't have to wait for it, but we can't create two of them at the same time

Reply:
https://issues.hibernatingrhinos.com/issue/RavenDB-15021